src/build_tools.rs

```diff
@@ -248,3 +248,33 @@ pub fn safe_repr(v: &PyAny) -> Cow<str> {
         "<unprintable object>".into()
     }
 }
+
+#[derive(Debug, Clone)]
+pub(crate) enum ExtraBehavior {
+    Unset,
+    Allow,
+    Forbid,
+    Ignore,
+}
+
+impl ExtraBehavior {
+    pub fn from_schema_or_config(py: Python, schema: &PyDict, config: Option<&PyDict>) -> PyResult<Self> {
```

Comment:
```suggestion
    pub fn from_schema_or_config(py: Python, schema: &PyDict, config: Option<&PyDict>, default: Self) -> PyResult<Self> {
```

Then you can remove the `Unset` option.

Reply:
Removed